nemo/collections/tts/modules/waveglow.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from enum import Enum
+
+import torch
+
+from nemo.collections.tts.modules.submodules import Invertible1x1Conv, WaveNet
+from nemo.core.classes import NeuralModule, typecheck
+from nemo.core.neural_types.elements import (
+    AudioSignal,
+    IntType,
+    MelSpectrogramType,
+    NormalDistributionSamplesType,
+    VoidType,
+)
+from nemo.core.neural_types.neural_type import NeuralType
+from nemo.utils.decorators import experimental
+
+
+class OperationMode(Enum):
```

Comment:
Models should be unnecessary after the function override of typecheck if you would like to refactor, but this is fine too.